src/renderers/dom/client/eventPlugins/__tests__/ChangeEventPlugin-test.js

```diff
@@ -12,7 +12,18 @@
 'use strict';
 
 var React = require('React');
+var ReactDOM = require('ReactDOM');
 var ReactTestUtils = require('ReactTestUtils');
+var ChangeEventPlugin = require('ChangeEventPlugin');
+
+var lastInputValue = ChangeEventPlugin.__lastInputValue;
+
+function setUntrackedValue(elem, value) {
```

Comment:
Deduping the events and only firing when the value changes creates a new layer of event triggering concerns.

Generally in a test (or where ever) I think that the expected behavior of manually triggering an event is that that event is fired regardless of the underlying logic. Since we are creating a new boundry of when these events are "really" fired that diverges from the browser, we run into the issue of not being able to just let manually triggered events pass through. I think the least disruptive approach is to try and maintain that behavior. 

One solution would be flagging events in `simulateNativeEventOnNode` and then having the plugin check for the flag to determine if it should compare last and current values or just flush the event through. That feels kinda gross to me, but the current state of manually resetting the cache is also pretty damn ugly outside of the plugin and plugin tests.